app/controllers/posts_controller.rb

```diff
@@ -1,16 +1,41 @@
 class PostsController < ApplicationController
+  before_action :authenticate_user!, only: [:new, :create]
+
 	def index
+    @categories = Category.all
 	end
 
-	def about
-	end
+  def new
+    @section = Section.find_by_id(params[:section_id])
 
-	def contacts
-	end
+    if @section.nil?
+      redirect_to posts_path
+    else
+      @post = Post.new
+    end
+  end
 
-	def press
-	end
+  def create
+    @section = Section.find_by_id(params[:section_id])
+
+    unless @section.nil?
+      @post = @section.posts.create(post_params)
+    end      
+      redirect_to posts_path
+  end
+
+  def show
+    @section = Section.find_by_id(params[:section_id])
```

Comment:
I suggest not passing through the `section_id` through the params.  Instead, I'd suggest having the route just be:

```
/posts/:id
```

Since the only thing that passing the section_id does is makes a second thing we have to look up (but we can always to `@post.section` to get the section)